managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -240,7 +240,6 @@ protected void recoverFromLedger(final ManagedCursorInfo info, final VoidCallbac
             }
 
             // Read the last entry in the ledger
-            cursorLedger = lh;
```

Comment:
very good point. I removed it because it is confusing to use the same variable to store 2 different kinds of ledger handler (one read-write and the other read-only)

Anyway I think it still make sense for the internal stats. In this case we would see:
```
cursorLedger = -1
cursorLedgerLastEntry = -1
```
and that is appropriate since we are in `NoLedger` state.

This below is the code that returns the `-1`:

```java
  public long getCursorLedger() {
        LedgerHandle lh = cursorLedger;
        return lh != null ? lh.getId() : -1;
    }

    public long getCursorLedgerLastEntry() {
        LedgerHandle lh = cursorLedger;
        return lh != null ? lh.getLastAddConfirmed() : -1;
    }
```